From: Tanu Kaskinen <tanu.kaskinen@linux.intel.com>
Date: Sat, 12 Jul 2014 18:59:19 +0300
Subject: path-lookup: don't make ~/.local/share/systemd/user a symlink

We already encourage upstreams to keep the default configuration
separate from user customizations for software that is installed in
the system location. Let's allow that separation also for software
that is installed in the home directory.

Some discussion:
http://thread.gmane.org/gmane.comp.sysutils.systemd.devel/19627

(cherry picked from commit 667a1cd645c9402921dedff08a86bb35cddbcbf9)
---
 src/shared/path-lookup.c | 18 ------------------
 1 file changed, 18 deletions(-)

diff --git a/src/shared/path-lookup.c b/src/shared/path-lookup.c
index e0aaf44..7d53d85 100644
--- a/src/shared/path-lookup.c
+++ b/src/shared/path-lookup.c
@@ -125,26 +125,8 @@ static char** user_dirs(
                         goto fail;
 
         } else if (home) {
-                _cleanup_free_ char *data_home_parent = NULL;
-
                 if (asprintf(&data_home, "%s/.local/share/systemd/user", home) < 0)
                         goto fail;
-
-                /* There is really no need for two unit dirs in $HOME,
-                 * except to be fully compliant with the XDG spec. We
-                 * now try to link the two dirs, so that we can
-                 * minimize disk seeks a little. Further down we'll
-                 * then filter out this link, if it is actually is
-                 * one. */
-
-                if (path_get_parent(data_home, &data_home_parent) >= 0) {
-                        _cleanup_free_ char *config_home_relative = NULL;
-
-                        if (path_make_relative(data_home_parent, config_home, &config_home_relative) >= 0) {
-                                mkdir_parents_label(data_home, 0777);
-                                (void) symlink(config_home_relative, data_home);
-                        }
-                }
         }
 
         e = getenv("XDG_DATA_DIRS");
